### PR TITLE
fix(Core/Quest): improved Overwhelmed!

### PR DESCRIPTION
**Pull request by grimgravy**
## Changes Proposed:
-  Fix: animation and movement event
-  Fix: added conditions that work only by the quest
-  Fix: cleave
-  Fix: hamstring
-  Fix: Mortal Strike 

## SOURCE: 

The quest is available from the wowhead site [overwhelmed](https://www.wowhead.com/quest=12288/overwhelmed)
the abilities [Wounded Skirmisher](https://www.wowhead.com/npc=27463/wounded-skirmisher#abilities)
-  Video 1 [youtube Overwhelmed](https://www.youtube.com/watch?v=GO0kMOjKWDo)
-  Video 2 [youtube Overwhelmed](https://www.youtube.com/watch?v=YCK8KkAwxJI)
## Tests Performed:

- was performed windows 7 64bits. Not a single error was found and it is Functional

## How to Test the Changes:

1. the quest is in grizzly hall
2. go creature 102487
3. quest add 12288

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1643290179356130100.sql [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+DELETE FROM `waypoint_data` WHERE `id`=274632;
+INSERT INTO `waypoint_data` (`id`, `point`, `position_x`, `position_y`, `position_z`, `orientation`, `delay`, `move_type`, `action`, `action_chance`, `wpguid`) VALUES
+(274632, 1, 4184.53, -2909.76, 280.17, 0, 0, 1, 0, 100, 0);
+
+/* implementing both conditions */
+
+DELETE FROM `spell_script_names` WHERE `spell_id`=48812 AND `ScriptName`='spell_renew_skirmisher';
+INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
+(48812, 'spell_renew_skirmisher');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1643290179356130100.sql:60]: doesn't exist.

Comment (grimgravy) [reply data/sql/updates/pending_db_world/rev_1643290179356130100.sql:60]: I forgot the script

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 2]
```diff
@@ -569,6 +694,42 @@ class npc_wounded_skirmisher : public CreatureScript
...
+{
+    NPC_WOUNDED_SKIRMISHER = 27463
+};
+
+class spell_renew_skirmisher : public SpellScriptLoader
+{
+public:
+    spell_renew_skirmisher() : SpellScriptLoader("spell_renew_skirmisher") { }
+
+    class spell_renew_skirmisher_SpellScript : public SpellScript
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:707]: use new script method please.

RegisterSpellScript

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 3]
```diff
@@ -520,47 +536,156 @@ class npc_wounded_skirmisher : public CreatureScript
 
     struct npc_wounded_skirmisherAI : public ScriptedAI
```
Comment (Kitzunu) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:537]: same here. As you are rewriting most of it. Please turn into new model

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp (hunk 2 above)
Comment (Kitzunu) [reply src/server/scripts/Northrend/zone_grizzly_hills.cpp:707]: https://www.azerothcore.org/wiki/core-scripts#stand-alone-spellscripts-and-aurascripts

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp (hunk 3 above)
Comment (Kitzunu) [reply src/server/scripts/Northrend/zone_grizzly_hills.cpp:537]: https://www.azerothcore.org/wiki/core-scripts#creature-scripts

**Comment by Lynethris** [comment]
Tested and work.

**Comment by FrancescoBorzi** [comment]
@azerothcore/developers please review this PR

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 4]
```diff
@@ -84,102 +84,102 @@ class npc_emily : public CreatureScript
...
-                case 20:
+                }
+                break;
+            case 11:
+                if (Creature* Mrfloppy = ObjectAccessor::GetCreature(*me, _mrfloppyGUID))
+                    Mrfloppy->GetMotionMaster()->MoveFollow(me, PET_FOLLOW_DIST, PET_FOLLOW_ANGLE);
+                break;
+            case 17:
+                if (Creature* Mrfloppy = ObjectAccessor::GetCreature(*me, _mrfloppyGUID))
+                    Mrfloppy->GetMotionMaster()->MovePoint(0, me->GetPositionX(), me->GetPositionY(), me->GetPositionZ());
```
Comment (Winfidonarleyan) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:109]: brackets

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 5]
```diff
@@ -84,102 +84,102 @@ class npc_emily : public CreatureScript
...
+                }
+                me->SetWalk(false);
+                break;
+            case 25:
+                Talk(SAY_VICTORY4);
+                break;
+            case 27:
+                me->DisappearAndDie();
+                if (Creature* Mrfloppy = ObjectAccessor::GetCreature(*me, _mrfloppyGUID))
+                    Mrfloppy->DisappearAndDie();
```
Comment (Winfidonarleyan) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:181]: brackets

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 6]
```diff
@@ -506,66 +506,204 @@ enum Skirmisher
...
+{
+public:
+    PrepareSpellScript(spell_renew_skirmisher);
+
+    SpellCastResult CheckRequirement()
+    {
+        if (Unit* caster = GetCaster())
+            if (Creature* wounded = caster->FindNearestCreature(NPC_WOUNDED_SKIRMISHER, 5.0f))
+                if (!wounded->IsInCombat())
+                    return SPELL_CAST_OK;
```
Comment (Winfidonarleyan) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp:699]: brackets

**Review by Winfidonarleyan — APPROVED** [review]

**Pull request merged**